src/modules/navigator/gpsfailure.cpp

```diff
@@ -93,7 +101,12 @@ GpsFailure::on_active()
 			q.copyTo(att_sp.q_d);
 			att_sp.q_d_valid = true;
 
-			_att_sp_pub.publish(att_sp);
+            if (_att_sp_pub != nullptr){
+                orb_publish(_att_setpoint_id, _att_sp_pub, &att_sp);
+            } else if (_att_setpoint_id) {
+                _att_sp_pub = orb_advertise(_att_setpoint_id, &att_sp);
+
+            }
```

Comment:
Could we do the same change without going back to the "old" uorb API? So could we use `_att_sp_pub.publish(att_sp);` in both places here?